src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public abstract class Figure implements AreaCalculator {
```

Comment:
Don't use abstract classes to set behavior for classes. Consider whether Figure should be an abstract class or if an interface would be more appropriate for defining common behavior, especially if you're not providing any default implementation for methods in Figure.